libs/blocks/preflight/panels/accessibility/check-image-alt-text.js

```diff
@@ -0,0 +1,61 @@
+import { getUniqueSelector } from './helper.js';
+
+/**
+ * Check for valid alt text on <img> elements.
+ * @param {HTMLElement[]} elements - The DOM elements to check.
+ * @param {Object} config - Optional configuration object.
+ * @returns {Array} List of accessibility violations for images.
+ */
+export default function checkImageAltText(elements = [], config = {}) {
+  const { checks = [] } = config;
+
+  // Skip this check if 'altText' isn't enabled in the config
+  if (!checks.includes('altText')) {
+    return [];
+  }
+
+  const violations = [];
+  const images = elements.filter((el) => el.tagName && el.tagName.toLowerCase() === 'img');
+
+  images.forEach((img) => {
+    const alt = img.getAttribute('alt');
+
+    const role = (img.getAttribute('role') || '').toLowerCase();
+    const ariaHidden = img.getAttribute('aria-hidden') === 'true';
+    const isDecorative = role === 'presentation' || ariaHidden;
+
+    const style = window.getComputedStyle(img);
+    const isHidden = style.display === 'none' || style.visibility === 'hidden' || parseFloat(style.opacity) === 0;
+
+    if (isDecorative || isHidden) return;
+
+    // Missing alt attribute or value is incorrect ( numbers or too short)
+    if (alt === null) {
+      violations.push({
+        description: 'Image is missing an alt attribute.',
+        impact: 'serious',
+        id: 'image-alt',
+        help: 'Add an alt attribute that describes the image.',
+        helpUrl: 'https://www.w3.org/WAI/tutorials/images/',
+        nodes: [{
+          target: [getUniqueSelector(img)],
+          html: img.outerHTML,
+        }],
+      });
+    } else if (/^\d+$/.test(alt.trim()) || alt.trim().length < 3) {
```

Comment:
I'm usually a fan of trying to avoid chaining if/else statements, similar to using early returns.
If/else statements they usually get added on and long-term turn into huge if/if-else/if-else/else statements that are very hard to decipher
```js
if (condition) {
// ...
// continue / break / return
}
```